### PR TITLE
Documentation: Add guide for how to add an auth provider to the backend

### LINKED ISSUE #1141: Auth Backend: Document how to add a passport.js based provider
Following the newly created `github` provider, document the steps required and how the code is organized and what to think about.

### PR DESCRIPTION
**Pull request by marcuseide**
As usual with documentation, let me know if I should re-organize it to make it more clear, if something is missing or outright wrong or what I spelled wrong.

Fixes #1141

### CODE REVIEW

File: plugins/auth-backend/add-auth-provider.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+**`plugins/auth-backend/src/providers/providerA/index.ts`** is simply re-exporting the create method to be used for hooking the provider up to the backend.
+
+```ts
+export { createProviderAProvider } from './provider';
+```
+
+### Hook it up to the backend
+
+**`plugins/auth-backend/src/providers/config.ts`** The provider needs to be configured properly so you need to add it to the list of configured providers, all of which implements [AuthProviderConfig](#AuthProviderConfig):
```
Comment (freben) [inline plugins/auth-backend/add-auth-provider.md:65]: `all of which implement`

File: plugins/auth-backend/add-auth-provider.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+**`plugins/auth-backend/src/providers/factories.ts`** When the `auth-backend` starts it sets up routing for all the available providers by calling `createAuthProviderRouter` on each provider. You need to import the create method from the provider and add it to the factory:
+
+```ts
+import { createProviderAProvider } from './providerA';
+const factories: { [providerId: string]: AuthProviderFactory } = {
+  providerA: createProviderAProvider,
+};
+```
+
+By doing this `auth-backend` automatically adds the these endpoints:
```
Comment (freben) [inline plugins/auth-backend/add-auth-provider.md:85]: `adds these`

File: plugins/auth-backend/add-auth-provider.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
+# Adding authentication providers
```
[the remark this answers is not in the document]
Comment (marcuseide) [reply plugins/auth-backend/add-auth-provider.md]: Yes, I moved it to `/docs/auth`. We'll probably make a lot of changes to the documentation structure shortly too.

File: docs/auth/add-auth-provider.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+router.get('/auth/providerA/refresh'); // if supported
+```
+
+As you can see each endpoint is prefixed with both `/auth` and its provider name.
+
+### Test the new provider
+
+You can `curl -i localhost:7000/auth/providerA/start` and which should provide a `302` redirect with a `Location` header. Paste the url from that header into a web browser and you should be able to trigger the authorization flow.
+
+### To summarize
```
Comment (soapraj) [inline docs/auth/add-auth-provider.md:153]: would it better if this was at the top and we link to the different sections?

File: docs/auth/add-auth-provider.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+  }
+
+  async start() {}
+  async handler() {}
+}
+```
+
+#### Adding an non-OAuth based provider
+
+An non-`OAuth` based provider could implement [AuthProviderRouteHandlers](#AuthProviderRouteHandlers) instead.
```
Comment (soapraj) [inline docs/auth/add-auth-provider.md:56]: We should add a note saying the non-OAuth provider is in an experimental stage? and would depend on the auth provider that you choose to add?

File: docs/auth/add-auth-provider.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
+# Adding authentication providers
```
Comment (soapraj) [inline docs/auth/add-auth-provider.md:1]: +1 on the docs :)

File: docs/auth/add-auth-provider.md (hunk 4 above)
Comment (marcuseide) [reply docs/auth/add-auth-provider.md]: 👍

**Review by stefanalund — APPROVED** [review]

**Pull request merged**